addons/account/models/account_move.py

```diff
@@ -5894,12 +5894,12 @@ def _autopost_draft_entries(self):
 
         try:  # try posting in batch
             with self.env.cr.savepoint():
-                moves._post()
+                moves.with_context(skip_account_deprecation_check=True)._post()
```

Comment:
Sorry, I know that I suggested this as a possibility, but after thinking about it, this will only work for the auto-post cron, but not other crons like for EDI.

Would it not make sense to apply my second suggestion instead and bypassing the constraint when the user is `SUPERUSER_ID`?

Reply:
or do you mean at this check ? 
https://github.com/odoo/odoo/blob/e2df544991d095c468b7057ef7335e16be04cd64/addons/account/models/account_move.py#L5357